contracts/src/instances/eeth/EETHConversions.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IERC20 } from "../../interfaces/IERC20.sol";
+import { ILiquidityPool } from "etherfi/src/interfaces/ILiquidityPool.sol";
+import { FixedPointMath, ONE } from "../../libraries/FixedPointMath.sol";
+import { IeETH } from "etherfi/src/interfaces/IeETH.sol";
+
+/// @author DELV
+/// @title EETHConversions
+/// @notice The conversion logic for the EETH integration.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+library EETHConversions {
+    using FixedPointMath for uint256;
+    /// @dev Convert an amount of vault shares to an amount of base.
+    /// @param _liquidityPool The Etherfi liquidity pool contract.
+    /// @param _eETH The eETH contract.
+    /// @param _shareAmount The vault shares amount.
+    /// @return The base amount.
+    function convertToBase(
+        ILiquidityPool _liquidityPool,
+        IERC20 _eETH,
+        uint256 _shareAmount
+    ) internal view returns (uint256) {
+        uint256 totalShares = IeETH(address(_eETH)).totalShares();
+        if (totalShares == 0) {
+            return 0;
+        }
+
+        // This calculation matches the implementation of
+        // `amountForShare(uint256 _share)` found in the LiquidityPool
+        //  contract.
+        // NOTE: Round down so that the output is an underestimate.
+        return
+            _shareAmount.mulDown(_liquidityPool.getTotalPooledEther()).divDown(
```

Comment:
It looks like they are really doing a `mulDivDown` [here](https://github.com/etherfi-protocol/smart-contracts/blob/716fd013593c0a46f24b586db3849fa1d28cd989/src/LiquidityPool.sol#L599). `mulDivDown(a, b, c)` is just:

```
(a * b) / c
```

`a.mulDown(b).divDown(c)` is:

```
(((a * b) / 1e18) * 1e18) / c
```

Reply:
ya this was a fuck up.